### PR TITLE
Fix-CreateCharge_EditCharge

### PR DESCRIPTION
**Pull request by rrpawar96**
https://github.com/openMF/community-app/issues/3452

### CODE REVIEW

File: app/scripts/controllers/product/CreateChargeController.js [hunk 1]
```diff
@@ -116,12 +116,11 @@
...
                                 scope.showenablefreewithdrawal = false;
                                 scope.showenablepaymenttype = false;
-                            }
-                            }
-                            else {
                                 scope.showdatefield = false;
                                 scope.repeatEvery = false;
                             }
+                            }
+
```
Comment (rrpawar96) [inline app/scripts/controllers/product/CreateChargeController.js:123]: datefield and repeatEvery part was not visible in the suitable conditions, this is fix for it

File: app/scripts/controllers/product/EditChargeController.js [hunk 2]
```diff
@@ -100,13 +100,16 @@
...
                 };
                 console.log(data.incomeOrLiabilityAccount);
                 if(data.incomeOrLiabilityAccount){
                     scope.formData.incomeAccountId = data.incomeOrLiabilityAccount.id;   
-                } 
+                }
+
+                if(data.paymentTypeOptions){
+                    scope.formData.paymentTypeId= data.paymentTypeOptions.id;
+                }
```
Comment (rrpawar96) [inline app/scripts/controllers/product/EditChargeController.js:111]: fix for issue https://github.com/openMF/community-app/issues/3452

**Comment by rrpawar96** [comment]
@luckyman20  Please review this PR

**Comment by er-prince** [comment]
@rrpawar96  hi , i have tested this PR

Client Charge Editing -  Fixed & showing previous data

client charge

Savings Charge Editing -  Fixed & also showing due date
savings charge due date

Share Charge Editing -  Fixed 
share account charge

Loan Charge  Edit-  Still coming blank , no previous data returned . 

loan charge

**Comment by francisguchie** [comment]
@er-prince  thank you so much for this testing

**Comment by rrpawar96** [comment]
@er-prince Thanks for testing it thoroughly, Please retest it again

**Comment by er-prince** [comment]
@rrpawar96  Tested it & all fixed . Thank you for your hard work again. 

Screenshot -  
Screenshot 2022-01-19 at 10 26 52 PM

**Comment by er-prince** [comment]
@francisguchie you are welcome :)

**Comment by er-prince** [comment]
i have checked the functionality , its working now as expected  in all charges ( Client, Savings & Deposit, Share, Loan) . i think its need to be merged soon as the 21.07 release & also patch release having edit charge broken.

**Comment by er-prince** [comment]
hi , i have found a new bug

log -  
2022-02-02 18:44:32.015 ERROR 802 --- [nio-443-exec-12] o.s.b.w.servlet.support.ErrorPageFilter  : Forwarding to error page from request [/api/v1/charges] due to exception [java.time.DateTimeException: Illegal value for DayOfMonth field, value 31 is not valid for month FEBRUARY]




I have set Monthly Charges  on date 31,  system will charge 2 times in 6 months interval ,  from feb 1 charges page  & savings account opening template isn't opening . even cant create new savings account too , as same charges linked to product.

**Comment by rrpawar96** [comment]
> I have set Monthly Charges on date 31, system will charge 2 times in 6 months interval , from feb 1 charges page & savings account opening template isn't opening . even cant create new savings account too , as same charges linked to product.

Hi @er-prince , i think this is another separate issue, we will create a new ticket for this and respective new PR for that Ticket

**Comment by er-prince** [comment]
hi @rrpawar96  , yes thanks . do you want me to create a new issue ticket for this?

**Comment by rrpawar96** [comment]
@er-prince, yes in https://issues.apache.org/jira/projects/FINERACT/issues, same like other issues

**Comment by rrpawar96** [comment]
Hi @luckyman20, Can you please review this pr, fix of edit charge page.

File: app/scripts/controllers/product/EditChargeController.js [hunk 3]
```diff
@@ -100,13 +100,15 @@
...
                     amount: data.amount
                 };
-                console.log(data.incomeOrLiabilityAccount);
                 if(data.incomeOrLiabilityAccount){
                     scope.formData.incomeAccountId = data.incomeOrLiabilityAccount.id;   
-                } 
+                }
+
+                if(data.paymentTypeOptions){
+                    scope.formData.paymentTypeId= data.paymentTypeOptions.id;
```
Comment (luckyman20) [inline app/scripts/controllers/product/EditChargeController.js]:
```suggestion
                    scope.formData.paymentTypeId = data.paymentTypeOptions.id;
```

**Review by luckyman20 — APPROVED** [review]
The rest looks good!

File: app/scripts/controllers/product/EditChargeController.js (hunk 3 above)
Comment (rrpawar96) [reply app/scripts/controllers/product/EditChargeController.js]: Comment addressed.

**Comment by rrpawar96** [comment]
@bharathcgowda, Please merge this PR

**Pull request merged**